servant/src/Servant/API/BoundedNat.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Servant.API.BoundedNat (BoundedNat (unBoundedNat), unsafeBoundedNat) where
+
+import           Control.Monad    (unless)
+import           Data.Aeson
+import           Data.Aeson.Types (modifyFailure)
+import           Data.Bifunctor   (bimap)
+import           Data.Ix
+import           Data.Proxy       (Proxy (Proxy))
+import qualified Data.Text        as T
+import           GHC.Generics     (Generic)
+import           GHC.TypeLits
+import           Servant.API
+
+-- | A newtype wrapper around 'Word' that ensures the value is within a given range.
+--
+-- Example:
+--
+-- >>> :set -XDataKinds
+-- >>> import Data.Proxy
+-- >>> let val = unsafeBoundedNat 5 :: BoundedNat 1 10
+-- >>> unBoundedNat val
+-- 5
+-- >>> let invalidVal = unsafeBoundedNat 15 :: BoundedNat 1 10
+-- >>> unBoundedNat invalidVal
+-- 15
+-- >>> checkBoundedRange (const $ fail "Out of range") val
+-- MkBoundedNat {unBoundedNat = 5}
+-- >>> checkBoundedRange (const $ fail "Out of range") invalidVal
+-- *** Exception: user error (Out of range)
+
+newtype BoundedNat (min :: Nat) (max :: Nat) = MkBoundedNat {unBoundedNat :: Word}
```

Comment:
Could it be renamed to `Range`? Unfortunately `BoundedNat` is more of a description than a name.